db/test/fns/queue_test.js

```diff
@@ -10,74 +10,218 @@ const taskcluster = require('taskcluster-client');
 suite(testing.suiteName(), function() {
   helper.withDbForProcs({ serviceName: 'queue' });
 
-  suite('message queue', function() {
-    setup('reset table', async function() {
+  suite('tests for pending tasks', function() {
+    setup('reset table', async function () {
       await helper.withDbClient(async client => {
-        await client.query('delete from azure_queue_messages');
+        await client.query('delete from queue_pending_tasks');
       });
     });
+    helper.dbTest('count empty queue', async function (db) {
+      assert.deepEqual(
+        await db.fns.queue_pending_tasks_count("tq1"),
+        [{ queue_pending_tasks_count: 0 }],
+      );
+    });
+    helper.dbTest('count queue containing messages', async function (db) {
+      await db.fns.queue_pending_tasks_put('tq1', 9, 'task1', 0, 'hint1', fromNow('10 seconds'));
+      await db.fns.queue_pending_tasks_put('tq1', 3, 'task2', 0, 'hint1', fromNow('10 seconds'));
+      await db.fns.queue_pending_tasks_put('tq1', 1, 'expiredTask', 0, 'hint1', fromNow('0 seconds'));
+      assert.deepEqual(
+        await db.fns.queue_pending_tasks_count("tq1"),
+        [{ queue_pending_tasks_count: 2 }],
+      );
+    });
+
+    // TODO: This requires support from the taskcluster-lib-postgres library
```

Comment:
I will do this in a follow up PR, to keep this strictly to the data migration and not build too many new features on top